provisioning/miniprov/hare_mp/main.py

```diff
@@ -908,7 +908,7 @@ def list2dict(
     for node in nodes_data_hctl:
         node_svc_info: Dict[str, List[str]] = {}
         for service in node['svcs']:
-            if not service['name'] in node_svc_info.keys():
```

Comment:
why to remove `.keys()`

Can you test manually this changes, as it not getting covered by any PR checks.

Reply:
@pavankrishnat  the default search in dict is in keys and time complexity is O(1).  Converting into list every time for each service is O(n) operation.  We can add a test or test manually.